modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/dao/DetectorMappingService.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.modelservice.dao;
+
+import com.expedia.adaptivealerting.modelservice.model.CreateDetectorMappingRequest;
+import com.expedia.adaptivealerting.modelservice.model.DetectorMapping;
+import com.expedia.adaptivealerting.modelservice.model.MatchingDetectorsResponse;
+import com.expedia.adaptivealerting.modelservice.model.SearchMappingsRequest;
+import com.expedia.adaptivealerting.modelservice.model.UpdateDetectorMappingRequest;
+
+import java.util.List;
+import java.util.Map;
+
+public interface DetectorMappingService {
```

Comment:
Consider using Spring Data Elasticsearch. (If not, I'd like to understand why not.) If so then please move this to the repo package, and adopt the same conventions you see there already, most of which are Spring Data conventions that allow SD to autogenerate query method implementations.

Reply:
This is noted as a separate change.